tools/graph_loading/construct_graph.py

```diff
@@ -0,0 +1,680 @@
+        The dictionary that is shared among all processes and saves the parsed edge data.
+    """
+    data = read_file(in_file)
+    feat_data = process_features(data, feat_ops) if feat_ops is not None else {}
+    if label_conf is not None:
+        label_data = process_labels(data, label_conf)
+        for key, val in label_data.items():
+            feat_data[key] = val
+    src_ids = data[src_id_col]
+    dst_ids = data[dst_id_col]
+    assert node_id_map is not None
+    src_type, _, dst_type = edge_type
+    if src_type in node_id_map:
+        src_ids = np.array([node_id_map[src_type][sid] for sid in src_ids])
+    else:
+        assert np.issubdtype(src_ids.dtype, np.integer), \
+                "The source node Ids have to be integer."
+    if dst_type in node_id_map:
+        dst_ids = np.array([node_id_map[dst_type][did] for did in dst_ids])
+    else:
+        assert np.issubdtype(dst_ids.dtype, np.integer), \
+                "The destination node Ids have to be integer."
+    return_dict[file_idx] = (src_ids, dst_ids, feat_data)
+
+def create_id_map(ids):
+    """ Create ID map
+
+    This creates an ID map for the input IDs.
+
+    Parameters
+    ----------
+    ids : Numpy array
+        The input IDs
+
+    Returns
+    -------
+    dict : the key is the original ID and the value is the new ID.
+    """
+    return {id1: i for i, id1 in enumerate(ids)}
+
+def process_node_data(process_confs, remap_id):
+    """ Process node data
+
+    We need to process all node data before we can process edge data.
+    Processing node data will generate the ID mapping.
+
+    The node data of a node type is defined as follows:
+    {
+        "node_id_col":  "<column name>",
+        "node_type":    "<node type>",
+        "format":       {"name": "csv", "separator": ","},
+        "files":        ["<paths to files>", ...],
+        "features":     [
+            {
+                "feature_col":  ["<column name>", ...],
+                "feature_name": "<feature name>",
+                "data_type":    "<feature data type>",
+                "transform":    {"name": "<operator name>", ...}
+            },
+        ],
+        "labels":       [
+            {
+                "label_col":    "<column name>",
+                "task_type":    "<task type: e.g., classification>",
+                "split_type":   [0.8, 0.2, 0.0],
+                "custom_train": "<the file with node IDs in the train set>",
+                "custom_valid": "<the file with node IDs in the validation set>",
+                "custom_test":  "<the file with node IDs in the test set>",
+            },
+        ],
+    }
+
+    Parameters
+    ----------
+    process_confs: list of dicts
+        The configurations to process node data.
+    remap_id: bool
+        Whether or not to remap node IDs
+
+    Returns
+    -------
+    dict: node ID map
+    dict: node features.
+    """
+    node_data = {}
+    node_id_map = {}
+    for process_conf in process_confs:
```

Comment:
Each process_conf corresponds to one node type, right?

Reply:
yes